### PR TITLE
On login, check if sync button is null before invoking it.

### PR DESCRIPTION
**Pull request by phillipm**
I was getting a null pointer exception sometimes upon login. This makes sure the syncButton isn't null before having it invoke methods.

Any ideas as to why it is null sometimes?

### CODE REVIEW

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java [hunk 1]
```diff
@@ -1507,13 +1507,14 @@ private void setSyncText(Pair<Long, int[]> syncDetails, String syncTextKey) {
...
-            syncButton.setNotificationText(syncIndicator);
-            adapter.notifyDataSetChanged();
-
-        } else {
-            if (syncButton != null) syncButton.setText(this.localize(syncTextKey));
+        if (syncButton != null) {
+            if (syncDetails.second[0] > 0) {
+                Spannable syncIndicator = (this.localize("home.sync.indicator", new String[]{String.valueOf(syncDetails.second[0]), Localization.get(syncTextKey)}));
+                syncButton.setNotificationText(syncIndicator);
+                adapter.notifyDataSetChanged();
```
Comment (phillipm) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: Its okay that notifyDataSetChanged isn't called when syncButton is null, right?

Comment (dcluna) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: Yea, but syncButton should not be null when calling this...

Comment (phillipm) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: When calling the setSyncText method? If so I will move this check to the caller of setSyncText

Comment (dcluna) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: Exactly, it only makes sense to set the syncButton text if it has been created (i.e., is visible)

**Comment by phillipm** [comment]
@dcluna updated

**Pull request merged**